File: sample-scala/app/views/util/Calc.scala
package views.util

import models.PaperSizeAndWhiteSpace

object Calc {

  def labelWidth(paper: PaperSizeAndWhiteSpace, cols: Int): Double = {
    (paper.width - 2 * paper.margin - (cols - 1) * paper.interLabelGutter) / cols.toDouble
  }

  def labelHeight(paper: PaperSizeAndWhiteSpace, rows: Int): Double = {
    (paper.height - 2 * paper.margin - (rows - 1) * paper.interLabelGutter) / rows.toDouble
  }
}

File: sample-java/app/views/util/Calc.java
package views.util;

import models.PaperSizeAndWhiteSpace;

public class Calc {

    public static double getLabelWidth(PaperSizeAndWhiteSpace paper, int cols) {
        return (paper.getWidth() - 2 * paper.getMargin() - (cols - 1) * paper.getInterLabelGutter())/(double)cols;
    }

    public static double getLabelHeight(PaperSizeAndWhiteSpace paper, int rows) {
        return (paper.getHeight() - 2 * paper.getMargin() - (rows - 1) * paper.getInterLabelGutter())/(double)rows;
    }
}
